### ISSUE TITLE
0.30 client results in UI warning "A required debug information file was missing"

### ISSUE THREAD

**Issue by pjenvey** [getsentry/sentry-rust#574]
### Environment

Linux, sentry-rust 0.30 w/ Cloud Sentry, Rust 1.68

### Steps to Reproduce

Upgrading from sentry 0.27 to 0.30 results in a new warning appearing in the Sentry UI: "A required debug information file was missing".
<img width="688" alt="Screenshot 2023-05-03 at 3 33 14 PM" src="https://user-images.githubusercontent.com/14639/236066473-f4afcd58-616c-459b-b44b-148ecc3f02fa.png">
This also produces an email alert stating "Failed to Process Events - Some events failed to process in your project".

With 0.27 we do see a note on each frame that it could not be symbolicated:
<img width="782" alt="Screenshot 2023-05-03 at 3 36 42 PM" src="https://user-images.githubusercontent.com/14639/236066556-86a18c38-52fc-4ab2-9bd3-0d1e8bb5a185.png">

With 0.30 this note has slightly changed to show:
<img width="765" alt="Screenshot 2023-05-03 at 3 36 49 PM" src="https://user-images.githubusercontent.com/14639/236066596-9cda2d16-4ab0-43ae-b7c5-0f491b1e9d81.png">

Otherwise the event appears as it normally does.

We've built this application with: 

```
[profile.release]
# Enables line numbers in Sentry
debug = 1
```

which has always caused our stack frame information to be included in sentry as seen in both screenshots.

What has began causing this error? I've never needed to e.g. [upload separate debug information](https://docs.sentry.io/product/cli/dif/) files before to sentry, compiling w/ [debug = 1](https://doc.rust-lang.org/cargo/reference/profiles.html) has always been good enough.

I do see the `debug-images` feature is now enabled by default as of 0.28, could this be related (I'm not entirely sure what it does)?

**Comment by pjenvey** [comment]
I've confirmed disabling `debug-images` makes this error go away. How should this work/the problem be avoided w/ the default feature set including `debug-images`?

**Comment by Swatinem** [comment]
Yes you are absolutely right, the `debug-images` integration is causing this problem.

If you are building *and shipping* with debug info included in your binary, you do not need the integration, and enabling it will either cause the problem you described, or worse, it can cause https://github.com/getsentry/sentry/issues/46435.

However, the thinking was that most projects are shipped with split / stripped debug info to production or to external users. In that case, you do need the `debug-images` integration, and you do need to upload that split debug info to Sentry, otherwise the stack traces will be useless.

This is not an obvious choice to make, as it depends on the specific build configuration, and I believe we can’t make features conditional on these build flags either. Not to mention that in both cases you are building with debug info, its just at shipping time when it is stripped.

**Comment by pjenvey** [comment]
Thanks @Swatinem. I can close this issue but maybe this should be left open to improve the documentation around the usage of debug information.

**Comment by smeubank** [comment]
https://github.com/getsentry/sentry-docs/pull/12104

Updated docs to call this out on the main rust docs page
